model/search_params_test.go

```diff
@@ -12,924 +12,1500 @@ import (
+			Input: "-#apple #banana from:chan",
+			Words: []searchWord{
+				{
+					value:   "#apple",
+					exclude: true,
+				},
+				{
+					value:   "#banana",
+					exclude: false,
+				},
+			},
+			Flags: []flag{
+				{
+					name:  "from",
+					value: "chan",
+				},
+			},
+		},
+		{
+			Name:  "string is multiple words containing a : prefixed with - and multiple # prefixes",
+			Input: "#apple #banana -from:chan",
+			Words: []searchWord{
+				{
+					value:   "#apple",
+					exclude: false,
+				},
+				{
+					value:   "#banana",
+					exclude: false,
+				},
+			},
+			Flags: []flag{
+				{
+					name:    "from",
+					value:   "chan",
+					exclude: true,
+				},
+			},
+		},
+		{
+			Name:  "string is multiple words containing a : prefixed with multiple - and multiple # prefixes",
+			Input: "-#apple -#banana -from:chan",
+			Words: []searchWord{
+				{
+					value:   "#apple",
+					exclude: true,
+				},
+				{
+					value:   "#banana",
+					exclude: true,
+				},
+			},
+			Flags: []flag{
+				{
+					name:    "from",
+					value:   "chan",
+					exclude: true,
+				},
+			},
+		},
+		{
+			Name:  "string is multiple words containing a : with a space",
+			Input: "apple banana from: chan",
+			Words: []searchWord{
+				{
+					value:   "apple",
+					exclude: false,
+				},
+				{
+					value:   "banana",
+					exclude: false,
+				},
+			},
+			Flags: []flag{
+				{
+					name:  "from",
+					value: "chan",
+				},
+			},
+		},
+		{
+			Name:  "string is multiple words containing a in: with a space",
+			Input: "apple banana in: chan",
+			Words: []searchWord{
+				{
+					value:   "apple",
+					exclude: false,
+				},
+				{
+					value:   "banana",
+					exclude: false,
+				},
+			},
+			Flags: []flag{
+				{
+					name:  "in",
+					value: "chan",
+				},
+			},
+		},
+		{
+			Name:  "string is multiple words containing a channel: with a space",
+			Input: "apple banana channel: chan",
+			Words: []searchWord{
+				{
+					value:   "apple",
+					exclude: false,
+				},
+				{
+					value:   "banana",
+					exclude: false,
+				},
+			},
+			Flags: []flag{
+				{
+					name:  "channel",
+					value: "chan",
+				},
+			},
+		},
+		{
+			Name:  "string is only fruit: with a space",
+			Input: "fruit: cherry",
+			Words: []searchWord{
+				{
+					value:   "fruit",
+					exclude: false,
+				},
+				{
+					value:   "cherry",
+					exclude: false,
+				},
+			},
+			Flags: []flag{},
+		},
+		{
+			Name:  "string is a single word with a : postfix",
+			Input: "channel:",
+			Words: []searchWord{
+				{
+					value:   "channel",
+					exclude: false,
+				},
+			},
+			Flags: []flag{},
+		},
+		{
+			Name:  "string is a single word which results in a single flag",
+			Input: "channel:first",
+			Words: []searchWord{},
+			Flags: []flag{
+				{
+					name:  "channel",
+					value: "first",
+				},
+			},
+		},
+		{
+			Name:  "string is a single word prefixed with - which results in a single excluded flag",
```

Comment:
`single flag with - which results in a excluded flag`

Reply:
We can follow this same approach in the next cases where we are talking about flags